internal/examples/agent/agent/agent.go

```diff
@@ -354,3 +394,60 @@ func (agent *Agent) onMessage(ctx context.Context, msg *types.MessageData) {
 		}
 	}
 }
+
+func (agent *Agent) onOpampConnectionSettings(ctx context.Context, settings *protobufs.OpAMPConnectionSettings) error {
+	if settings == nil || settings.Certificate == nil {
+		agent.logger.Debugf("Received nil certificate offer, ignoring.\n")
+	}
+
+	cert, err := agent.getCertFromSettings(settings.Certificate)
+	if err != nil {
+		return err
+	}
+
+	agent.logger.Debugf("Reconnecting to verify offered client certificate.\n")
+
+	agent.disconnect()
+	agent.opampClientCert = cert
+	if err := agent.connect(); err != nil {
```

Comment:
So, yes, correct, Stop()/disconnect() will return immediately, a new client will be created by connect() and will connect while the previous client is still shutting down. I think it is fine, I don't see a problem with this.

Alternately, we can wait until stopping is done and then attempt the new connection, but in this example I think it doesn't matter.

>It could be possible that SetRemoteConfigStatus is sent to the new server instead of the prev server which offered the updated config.

In this example we don't really change the destination server, so I don't think this is an issue (should not cause any problems on the server side of our example). In a production implementation I think you are right you would go with complete stopping (waiting for it) and the initiating the new connection. 

This would make the code more complicated. It is doable, perhaps can be done in a separate PR, I don't want to make already large PR even larger.

Reply:
Right, I was trying to understand how would this work in practice but this is fine for example

Reply:
Thinking more about this I believe you are right, we need to do better in the example. I added an issue to address this in a future PR: https://github.com/open-telemetry/opamp-go/issues/178